zeg/imagesets.py

```diff
@@ -36,42 +38,167 @@ def get(log, session, args):
     log.warn('Get imageset command coming soon.')
 
 
+def _get_chunk_upload_futures(
+    executor, paths, session, create_url, complete_url, log, workload_size
+):
+    """Return executable tasks with image uploads in batches.
+
+    Instead of performing image uploads and updating the imageset
+    one at a time, we reduce load on the API server by uploading
+    many images and updating the api server in one go, consequently
+    making upload faster.
+    """
+    total_work = len(paths)
+    workloads = []
+    temp = []
+
+    i = 0
+    while i < total_work:
+        path = paths[i]
+        temp.append(path)
+        i += 1
+        if len(temp) == workload_size or i == total_work:
+            workload_info = {
+                "start": i - len(temp),
+                "count": len(temp),
+            }
```

Comment:
I found this a bit tricky to follow. There might be clearer ways - just thinking out loud below with help from SO.
```
def chunks(path, stride):
    """Given paths, break into chunks as dict consisting of the subset list, and start index."""
    for i in range(0, len(paths), stride):
        yield {"items": paths[i:i + stride], "start": i}

for chunk in chunks(paths, workload_size):
    workload_info = {
        "start": chunk["start"],
        "count": len(chunk["items"])
    }

    workloads.append(...)
```